pkg/utils/nodes.go

```diff
@@ -5,15 +5,22 @@ import (
 
 	corev1 "k8s.io/api/core/v1"
 	apiErrors "k8s.io/apimachinery/pkg/api/errors"
-	"k8s.io/apimachinery/pkg/types"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
+// getNodeWithName returns a node with a name nodeName, or an error if it can't be found
+func getNodeWithName(r client.Reader, nodeName string) (*corev1.Node, error) {
+	node := &corev1.Node{}
+	key := client.ObjectKey{Name: nodeName}
+	if err := r.Get(context.TODO(), key, node); err != nil {
```

Comment:
you can just return `r.Get(...)`

Reply:
Do you mean?
```suggestion
     return node, r.Get(context.TODO(), key, node)
```

Reply:
Ah sorry, somehow I thought the node is also a return value of Get()
But this should work as well.
Even though it looks a bit strange 🙈 
Leaving it up to you if you want to keep this or revert.

Reply:
I will revert it 👍🏻